src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp

```diff
@@ -103,125 +106,132 @@ namespace AirflowNetwork {
     };
 
-    int constexpr PressureCtrlExhaust = 1;
-    int constexpr PressureCtrlRelief = 2;
+    enum class PressureCtrl
+    {
+        Unassigned = -1,
+        Exhaust,
+        Relief
+    };
 
-    // DERIVED TYPE DEFINITIONS:
+    enum class AirflowNetworkControl
+    {
+        Unassigned = -1,
+        Simple,    // Simple calculations only
+        Multizone, // Perform multizone calculations only
+        NoneADS,   // Perform distribution system during system on time only
+        SimpleADS, // Perform distribution system during system on time and simple calculations during off time
+        MultiADS   // Perform distribution system during system on time and multizone calculations during off time
+
+    };
 
-    // MODULE VARIABLE DECLARATIONS:
     // Node simulation variable in air distribution system
     // Link simulation variable in air distribution system
     // Sensible and latent exchange variable in air distribution system
 
-    // Vent Control  DistSys Control  Flag    Description
-    //  NONE           NONE           0      No AirflowNetwork and SIMPLE
-    //  SIMPLE         NONE           1      Simple calculations only
-    //  MULTIZONE      NONE           2      Perform multizone calculations only
-    //  NONE           DISTSYS        3      Perform distribution system during system on time only
-    //  SIMPLE         DISTSYS        4      Perform distribution system during system on time and simple calculations during off time
-    //  MULTIZONE      DISTSYS        5      Perform distribution system during system on time and multizone calculations during off time
-
-    int constexpr AirflowNetworkControlSimple(1);    // Simple calculations only
-    int constexpr AirflowNetworkControlMultizone(2); // Perform multizone calculations only
-    int constexpr AirflowNetworkControlSimpleADS(4); // Perform distribution system during system
-    // on time and simple calculations during off time
-    int constexpr AirflowNetworkControlMultiADS(5); // Perform distribution system during system on time
-                                                    // and multizone calculations during off time
+    // Vent Control  DistSys Control     Description
+    //  NONE           NONE             No AirflowNetwork and SIMPLE
+    //  SIMPLE         NONE             Simple calculations only
+    //  MULTIZONE      NONE             Perform multizone calculations only
+    //  NONE           DISTSYS          Perform distribution system during system on time only
+    //  SIMPLE         DISTSYS          Perform distribution system during system on time and simple calculations during off time
+    //  MULTIZONE      DISTSYS          Perform distribution system during system on time and multizone calculations during off time
 
     // Types
 
     struct AirflowNetworkSimuProp // Basic parameters for AirflowNetwork simulation
     {
         enum class Solver
         {
-            SkylineLU,
+            Unassigned = -1,
```

Comment:
Please remove Unassigned

Reply:
@jasondegraw, we are going to create a new pattern for all enumerated values that searches `constexpr std::array<std::string_view>` to map string names to enumerated values. That singular lookup function will return -1 if the enumerated string is not found, and the idea was to use -1 as the constructor initialization value as well. I think you can ignore the Unassigned value for all intents and purposes. What problems do you think it will create?

Reply:
That pattern (which is not a good pattern at any rate) might work for some things where values can be "unassigned", but in all of the cases I'm pointing to here cannot be unassigned. It's got to be one of the valid choices, and there's a default already assigned. There are IDEs/compilers out there that will warn about not all enumeration values being accounted for in switch statements, so ignoring a superfluous value reinforces bad behavior.

It's a bad pattern because using magic values is always a bad pattern.  Always. It would be more proper C++ to return a `std::optional` and require the caller to take affirmative steps to obtain the value.

Reply:
Ummm, returning an index of -1 when an item (or a NULL if you are dealing with objects and not indices) is not found in list is not a "magic value", it's a pattern as old as C itself. `std::optional` may or may not be "more proper" C++ (is it really more proper for list searches to return an optional? really? why?) but it's also responsible for all of the get nonsense in OpenStudio measures. You can do whatever you want in AFN and you can propagate the `std::optional` pattern to the rest of EnergyPlus after I leave DOE.

Reply:
I know the OpenStudio experience with optionals has been... uh... difficult, but I'd argue that there was some overuse there. Maybe a lot of overuse. Dan probably agrees. Anyway, I'd say yes, returning an optional from a search would be a good use of an optional, at least until the "future of errors" presentations that I've seen become reality (static exceptions and all that jazz). It requires the caller to do something about what got returned and is not quite the blunt instrument that `[[nodiscard]]` is.

Reply:
Well, I would argue that until the future of errors is decided, -1 is better than `std::optional` for both sanity and the program critical path. And however distasteful and "bad" this pattern may seem, I am sure we can find at least 29,000 other worse ones to argue about.